code/standards/checkstyle.xml

```diff
@@ -75,13 +83,21 @@
     <module name="FinalClass"/>
     <module name="HideUtilityClassConstructor"/>
     <module name="InterfaceIsType"/>
-    <module name="VisibilityModifier"/>
+    <module name="VisibilityModifier">
+      <property name="protectedAllowed" value="true" />
+      <property name="allowPublicFinalFields" value="true" />
```

Comment:
Setting this to needs review - as we need a specific plan for mutable public final fields